jack_port_set_name: deprecated

With the latest version of Jack 2 (1.9.11, the latest git) I get this warning when loading drumkits in Hydrogen:
jack_port_set_name: deprecated

The same warning appears during compilation:
```
/home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp: In member function ‘void H2Core::JackOutput::setTrackOutput(int, H2Core::Instrument*, H2Core::InstrumentComponent*, H2Core::Song*)’:
/home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:705:2: warning: ‘int jack_port_set_name(jack_port_t*, const char*)’ is deprecated [-Wdeprecated-declarations]
  jack_port_set_name( track_output_ports_L[n], ( chName + "L" ).toLocal8Bit() );
  ^
In file included from /home/blablack/src/hydrogen/src/core/include/hydrogen/IO/JackOutput.h:34:0,
                 from /home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:23:
/usr/include/jack/jack.h:899:5: note: declared here
 int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_DEPRECATED_EXPORT;
     ^
/home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:705:2: warning: ‘int jack_port_set_name(jack_port_t*, const char*)’ is deprecated [-Wdeprecated-declarations]
  jack_port_set_name( track_output_ports_L[n], ( chName + "L" ).toLocal8Bit() );
  ^
In file included from /home/blablack/src/hydrogen/src/core/include/hydrogen/IO/JackOutput.h:34:0,
                 from /home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:23:
/usr/include/jack/jack.h:899:5: note: declared here
 int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_DEPRECATED_EXPORT;
     ^
/home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:705:78: warning: ‘int jack_port_set_name(jack_port_t*, const char*)’ is deprecated [-Wdeprecated-declarations]
  jack_port_set_name( track_output_ports_L[n], ( chName + "L" ).toLocal8Bit() );
                                                                              ^
In file included from /home/blablack/src/hydrogen/src/core/include/hydrogen/IO/JackOutput.h:34:0,
                 from /home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:23:
/usr/include/jack/jack.h:899:5: note: declared here
 int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_DEPRECATED_EXPORT;
     ^
/home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:706:2: warning: ‘int jack_port_set_name(jack_port_t*, const char*)’ is deprecated [-Wdeprecated-declarations]
  jack_port_set_name( track_output_ports_R[n], ( chName + "R" ).toLocal8Bit() );
  ^
In file included from /home/blablack/src/hydrogen/src/core/include/hydrogen/IO/JackOutput.h:34:0,
                 from /home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:23:
/usr/include/jack/jack.h:899:5: note: declared here
 int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_DEPRECATED_EXPORT;
     ^
/home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:706:2: warning: ‘int jack_port_set_name(jack_port_t*, const char*)’ is deprecated [-Wdeprecated-declarations]
  jack_port_set_name( track_output_ports_R[n], ( chName + "R" ).toLocal8Bit() );
  ^
In file included from /home/blablack/src/hydrogen/src/core/include/hydrogen/IO/JackOutput.h:34:0,
                 from /home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:23:
/usr/include/jack/jack.h:899:5: note: declared here
 int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_DEPRECATED_EXPORT;
     ^
/home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:706:78: warning: ‘int jack_port_set_name(jack_port_t*, const char*)’ is deprecated [-Wdeprecated-declarations]
  jack_port_set_name( track_output_ports_R[n], ( chName + "R" ).toLocal8Bit() );
                                                                              ^
In file included from /home/blablack/src/hydrogen/src/core/include/hydrogen/IO/JackOutput.h:34:0,
                 from /home/blablack/src/hydrogen/src/core/src/IO/jack_output.cpp:23:
/usr/include/jack/jack.h:899:5: note: declared here
 int jack_port_set_name (jack_port_t *port, const char *port_name) JACK_OPTIONAL_WEAK_DEPRECATED_EXPORT;
     ^
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,6 +203,7 @@ ELSE()
 ENDIF()
 
 FIND_HELPER(JACK jack jack/jack.h jack)
+CHECK_LIBRARY_EXISTS(jack jack_port_rename "" HAVE_JACK_PORT_RENAME)
 FIND_HELPER(JACKSESSION jack jack/session.h jack)
 IF(APPLE)
     FIND_LIBRARY(AUDIOUNIT_LIBRARY AudioUnit)

--- a/src/core/include/hydrogen/config.h.in
+++ b/src/core/include/hydrogen/config.h.in
@@ -31,6 +31,7 @@
 
 #cmakedefine HAVE_SSCANF
 #cmakedefine HAVE_RTCLOCK
+#cmakedefine HAVE_JACK_PORT_RENAME
 
 #ifndef H2CORE_HAVE_DEBUG
 #cmakedefine H2CORE_HAVE_DEBUG

--- a/src/core/src/IO/jack_output.cpp
+++ b/src/core/src/IO/jack_output.cpp
@@ -702,8 +702,13 @@ void JackOutput::setTrackOutput( int n, Instrument * instr, InstrumentComponent 
 	DrumkitComponent* p_dmCompo = song->get_component( compo->get_drumkit_componentID() );
 	chName = QString( "Track_%1_%2_%3_" ).arg( n + 1 ).arg( instr->get_name() ).arg( p_dmCompo->get_name() );
 
+#ifdef HAVE_JACK_PORT_RENAME
+	jack_port_rename( client, track_output_ports_L[n], ( chName + "L" ).toLocal8Bit() );
+	jack_port_rename( client, track_output_ports_R[n], ( chName + "R" ).toLocal8Bit() );
+#else
 	jack_port_set_name( track_output_ports_L[n], ( chName + "L" ).toLocal8Bit() );
 	jack_port_set_name( track_output_ports_R[n], ( chName + "R" ).toLocal8Bit() );
+#endif
 }
 
 void JackOutput::play()

